crypto/ec/ec_key.c

```diff
@@ -195,59 +195,95 @@ int ossl_ec_key_gen(EC_KEY *eckey)
     return eckey->group->meth->keygen(eckey);
 }
 
+/*
+ * ECC Key generation.
+ * See SP800-56AR3 5.6.1.2.2 "Key Pair Generation by Testing Candidates"
+ *
+ * Params:
+ *     eckey An EC key object that contains domain params. The generated keypair
+ *           is stored in this object.
+ * Returns 1 if the keypair was generated or 0 otherwise.
+ */
 int ec_key_simple_generate_key(EC_KEY *eckey)
 {
     int ok = 0;
     BN_CTX *ctx = NULL;
     BIGNUM *priv_key = NULL;
     const BIGNUM *order = NULL;
     EC_POINT *pub_key = NULL;
+    const EC_GROUP *group = eckey->group;
 
     if ((ctx = BN_CTX_new()) == NULL)
```

Comment:
While we are at it we could also make this a secure ctx with `BN_CTX_secure_new()`

Reply:
Looks like I can get rid of the ctx instead, since  it is only used by the EC_POINT_mul which handles passing NULL (and then used BN_CTX_secure_new() internally)

Reply:
Good point!